File: src/main/java/com/example/objects/pages/WebPage.java
package com.example.objects.pages;
import com.codeborne.selenide.WebDriverRunner;

import static com.codeborne.selenide.WebDriverRunner.*;
import static org.junit.jupiter.api.Assertions.*;

public class WebPage {

    public String accountPage;
    public String accouuntEmail;

    public String url;
    public String title;

    public WebPage shouldHaveUrl(String url){
        assertEquals(url, getWebDriver().getCurrentUrl(), "Page Url should be:" + url);
        return this;
    }

    public WebPage shouldContainUrl(String url){
        assertTrue(getWebDriver().getCurrentUrl().contains(url), "Page should containt url" + url);
        return this;
    }

    public WebPage shouldHaveTitle(String title){
        assertEquals(title, getWebDriver().getTitle(), "Page Title should be: "+ title);
        return this;
    }

    public WebPage shouldContainTitle(String title){
        assertTrue(getWebDriver().getTitle().contains(title), "Page should containt title" + title);
        return this;
    }
}

File: src/main/kotlin/com/exmple/electric_flow/WebPage.kt
package com.exmple.electric_flow

import com.codeborne.selenide.WebDriverRunner
import com.exmple.electric_flow.extensions.Extensions
import com.exmple.gepur.models.pages.WebPage
import org.junit.jupiter.api.Assertions


abstract class WebPage: Extensions() {

    abstract val url: String
    abstract val title: String



}